test/stash_test.exs

```diff
@@ -42,7 +42,7 @@ defmodule StashTest do
       stash = Stash.init() |> Stash.put("key1", "value1") |> Stash.put("key2", "value2")
       {new_stash, _removed_value} = stash |> Stash.delete("key1")
 
-      assert stash == [{"key2", "value2"}]
+      assert Enum.count(stash) == Enum.count(new_stash) + 1
```

Comment:
LGTM. It might be marginally better with `Enum.count(stash) - 1 == Enum.count(new_stash) ` which aligns more closely to the test name, but at this point its an exercise in pedantry.